src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -175,12 +177,13 @@ const MapPreview: React.FC<MapPreviewProps> = ({
             }}
             otherIconsStyle={{ opacity: 0.4 }}
             lockAspectRatio
-            isLoading={isLoading}
+            isSaving={isSaving}
           />
         )}
 
-        <S.EditButton disabled={isLoading} onClick={handleEditButton}>
-          {isLoading ? "Saving..." : isEditing ? "Save layout" : "Edit layout"}
+        <S.EditButton disabled={isSaving} onClick={handleEditButton}>
+          {isSaving && <div>Saving...</div>}
+          {!isSaving && <div>{editButtonText}</div>}
```

Comment:
[NB] This could stay as a ternary now that you've separated the `editButtonText` part out of it.